### PR TITLE
Add `SIMILAR TO` syntax and description

### LINKED ISSUE #1675: Document: feat(expr): implement `<expr> [ NOT ] SIMILAR TO <pat> [ ESCAPE <esc_text> ]`
This issue tracks the documentation update needed for the merged PR #14000.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/14000
Source PR Merged At: 2023-12-21T07:09:37Z

### PR DESCRIPTION
**Pull request by WanYixian**
## Info

- **Description**

  Add info about `SIMILAR TO` syntax

- **Note**

  This is a draft, feel free to update it if anything needs revision, thanks.

- **Related code PR**

  https://github.com/risingwavelabs/risingwave/pull/14000

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1675

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/functions-operators/sql-function-string.md [hunk 1]
```diff
@@ -681,3 +681,41 @@ You can use `ESCAPE ''` to disable the escape mechanism, but specifying a custom
...
+| {m,}     | Repeats the previous item m or more times.              |
+| {m,n}    | Repeats the previous item at least m and not more than n times. |
+| ()       | Parentheses group items into a single logical item.   |
+| [...]    | A bracket expression specifies a character class. |
+
+### Escape
+
+To match a metacharacter literally, use the escape character `\` before the respective character in the pattern. To match the escape character itself, write two escape characters: `\\`.
+
+You can use `ESCAPE ''` to disable the escape mechanism, but specifying a custom escape character using the `ESCAPE` clause is not supported.
```
Comment (jetjinser) [inline docs/sql/functions-operators/sql-function-string.md]: specifying a custom escape character using the `ESCAPE` clause in the `SIMILAR TO` clause is **supported**.
And the custom escape character must be empty or one character.

[e2e test](https://github.com/risingwavelabs/risingwave/blob/8c0a1a4fa68c7b484e81bc8709fb714de75b6b98/e2e_test/batch/functions/similar_to_escape.slt.part#L77-L95), [unit test](https://github.com/risingwavelabs/risingwave/blob/7954da3e7ea628d4d27d840287c31bd5148d2c9a/src/expr/impl/src/scalar/similar_to_escape.rs#L168) and [source code](https://github.com/risingwavelabs/risingwave/blob/7954da3e7ea628d4d27d840287c31bd5148d2c9a/src/expr/impl/src/scalar/similar_to_escape.rs#L22).

btw, [this comment](https://github.com/risingwavelabs/risingwave/blob/8c0a1a4fa68c7b484e81bc8709fb714de75b6b98/e2e_test/batch/functions/similar_to_escape.slt.part#L71) is a mistake, don't be confused, fix in risingwavelabs/risingwave#14393.

File: docs/sql/functions-operators/sql-function-string.md [hunk 2]
```diff
@@ -681,3 +681,41 @@ You can use `ESCAPE ''` to disable the escape mechanism, but specifying a custom
...
+
+### Examples
+
+```sql
+'abc' SIMILAR TO 'a|b|c'                false
+'abc' SIMILAR TO '(a|b|c)+'             true
```
Comment (ShanlanLi) [inline docs/sql/functions-operators/sql-function-string.md:723]: Do we need that small arrow to indicate what is the result?

**Review by ShanlanLi — APPROVED** [review]
LGTM

File: docs/sql/functions-operators/sql-function-string.md (hunk 1 above)
Comment (WanYixian) [reply docs/sql/functions-operators/sql-function-string.md]: Thanks for pointing this out. Revised now and pls take a look

File: docs/sql/functions-operators/sql-function-string.md (hunk 2 above)
Comment (WanYixian) [reply docs/sql/functions-operators/sql-function-string.md:723]: I also considered adding them before. Later removed the arrows to ensure consistency with the code example of [like pattern](https://docs.risingwave.com/docs/current/sql-function-string/#examples).

**Review by jetjinser — APPROVED** [review]
LGTM

File: docs/sql/functions-operators/sql-function-string.md [hunk 3]
```diff
@@ -681,3 +681,41 @@ You can use `ESCAPE ''` to disable the escape mechanism, but specifying a custom
 'abc' LIKE '_b_'           true
 'abc' LIKE 'c'             false
 ```
+
+## `SIMILAR TO` pattern matching expressions
+
+```sql
+string [ NOT ] SIMILAR TO pattern [ ESCAPE '' ]
```
Comment (wangrunji0408) [inline docs/sql/functions-operators/sql-function-string.md]:
```suggestion
string [ NOT ] SIMILAR TO pattern [ ESCAPE escape-character ]
```

**Review by wangrunji0408 — COMMENTED** [review]
Others LGTM!

**Review by wangrunji0408 — APPROVED** [review]

**Pull request merged**